apps/mobile/src/utils/notifications/cleanup.ts

```diff
@@ -0,0 +1,166 @@
+import { type Address } from '@/src/types/address'
+import { type RootState } from '@/src/store'
+import { selectAllDelegatesForSafeOwners } from '@/src/store/delegatesSlice'
+import Logger from '@/src/utils/logger'
+
+export interface NotificationCleanupError {
+  type: 'safe' | 'blocking'
+  message: string
+  originalError: unknown
+}
+
+export interface SafeNotificationInfo {
+  address: string
+  chainIds: string[]
+}
+
+/**
+ * Classifies errors to determine if they should block private key deletion
+ */
+export const classifyNotificationError = (error: unknown): NotificationCleanupError => {
+  if (error && typeof error === 'object' && 'status' in error) {
+    const status = (error as { status: number }).status
+
+    // Safe errors - don't block deletion
+    if (status === 404 || status === 410) {
+      return {
+        type: 'safe',
+        message: 'Subscription already removed',
+        originalError: error,
+      }
+    }
+
+    // Blocking errors - prevent deletion
+    if (status >= 500 || status === 401 || status === 403) {
+      return {
+        type: 'blocking',
+        message: `Server error (${status}): Cannot verify subscription removal`,
+        originalError: error,
+      }
+    }
+
+    // Rate limiting - blocking but with retry suggestion
+    if (status === 429) {
+      return {
+        type: 'blocking',
+        message: 'Rate limited: Too many requests. Please try again in a moment.',
+        originalError: error,
+      }
+    }
+  }
+
+  // Network errors, timeouts, etc. - blocking by default
+  return {
+    type: 'blocking',
+    message: 'Network error: Cannot verify subscription removal',
+    originalError: error,
+  }
+}
+
+/**
+ * Utility for retrying operations with exponential backoff
+ */
+export const withRetry = async <T>(operation: () => Promise<T>, maxRetries = 3, baseDelay = 1000): Promise<T> => {
```

Comment:
This seems very similar to `retryWithBackoff`. Do we need both?

Reply:
good point. I've consolidated those now to a single function.